### PR TITLE
Added Environment OS Platform version switch case

### PR DESCRIPTION
**Pull request by StephenHodgson**
# XRTK - Mixed Reality Toolkit Change Request

## Overview

High level overview of this pull request.

## Target of the change:

Is this enhancement for:

- Core
- Platform 

## Changes:

Brief list of the targeted features that are being changed.

- Fixes: #303

### CODE REVIEW

**Comment by StephenHodgson** [comment]
@greyfoxlinux can you confirm this fixed the symbolic linked folders being generated correctly?

File: XRTK-Core/Packages/com.xrtk.core/Inspectors/Utilities/GitUtilities.cs [hunk 1]
```diff
@@ -117,6 +117,7 @@ public static void WritePathToGitIgnore(string ignoredPath)
         internal static bool UpdateSubmodules()
         {
             EditorUtility.DisplayProgressBar("Updating Submodules...", "Please wait...", 0.5f);
+            // TODO This may also not work for Mac Editors.
             var success = new Process().Run($"/C cd \"{RepositoryRootDir}\" && git submodule update --init --all", out _);
```
Comment (StephenHodgson) [inline XRTK-Core/Packages/com.xrtk.core/Inspectors/Utilities/GitUtilities.cs]: @greyfoxlinux what would the mac terminal command equivalent be? do ya know?

File: XRTK-Core/Packages/com.xrtk.core/Inspectors/Utilities/SymbolicLinks/SymbolicLinker.cs [hunk 2]
```diff
@@ -374,11 +374,33 @@ private static bool CreateSymbolicLink(string sourceAbsolutePath, string targetA
...
 
-            // --------------------> /C mklink /D "C:\Link To Folder" "C:\Users\Name\Original Folder"
-            if (!new Process().Run($"/C mklink /D \"{targetAbsolutePath}\" \"{sourceAbsolutePath}\"", out var error))
+            switch (Environment.OSVersion.Platform)
             {
-                Debug.LogError($"{error}");
-                return false;
+                case PlatformID.MacOSX:
+                    // --------------------> ln -s /path/to/original /path/to/symlink
+                    if (!new Process().Run($"ln -s \"{sourceAbsolutePath.ToForwardSlashes()}\" \"{targetAbsolutePath.ToForwardSlashes()}\"", out var error))
```
Comment (StephenHodgson) [inline XRTK-Core/Packages/com.xrtk.core/Inspectors/Utilities/SymbolicLinks/SymbolicLinker.cs]: Need to add `"/Applications/Utilities/Terminal.app/Contents/MacOS/Terminal"`

and the terminal doesn't seem to work with relative paths.

File: XRTK-Core/Packages/com.xrtk.core/Inspectors/Utilities/SymbolicLinks/SymbolicLinker.cs [hunk 3]
```diff
@@ -394,7 +416,9 @@ private static bool DeleteSymbolicLink(string path)
                 return false;
             }
 
-            if (new Process().Run($"/C rmdir /q \"{path}\"", out _))
+            var success = new Process().Run($"/C rmdir /q \"{path}\"", out _, @"cmd.exe");
```
Comment (StephenHodgson) [inline XRTK-Core/Packages/com.xrtk.core/Inspectors/Utilities/SymbolicLinks/SymbolicLinker.cs]: need to finish this one

**Comment by SimonDarksideJ** [comment]
Is this actually ready for review, as you have some comments about finishing stuff @StephenHodgson

**Comment by StephenHodgson** [comment]
Okay this is ready to go. Tested on my end. Not sure who else has a Mac to pull this with.

**Pull request merged**